quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -1079,14 +1081,18 @@ def _remove_empty_dirs(self):
         for root, dirs, __ in os.walk(self.expanded_destination, topdown=False):
             for dirname in dirs:
                 dir_path = os.path.realpath(os.path.join(root, dirname))
-                if not os.listdir(dir_path):
+                files = os.listdir(dir_path)
+                last_file_is_cover = files and files[0] == 'cover.jpg'
```

Comment:
That's the _first_ file, no? What if there are lots and `'cover.jpg'` just happens to be the first?

Reply:
If there are other files than the music and `cover.jpg`, deleting the folder will fail. And I think that is okay, since the user shall notice that's not the way how this plugin sync process is designed. Then an error message explaining that would be nice I think.

Reply:
Ah, yeah that makes sense. Perhaps you could add a comment to explain this in the code...